pkg/proto/nbcontract/v1/nodepoolprofile.proto

```diff
@@ -0,0 +1,7 @@
+syntax = "proto3";
+package nbcontract.v1;
+
+message NodePoolProfile {
+  // vm_size represents the size of this node pool
+  string vm_size = 1;
+}
```

Comment:
This does not quite make sense to me. For one thing, I assume vm_size is actually the VM SKU (like Standard_D2_v5), rather than the size of the node pool. For another, it is not clear to me why VHD/bootstrap would need to know either (and it could likely find out the VM SKU by itself, if it is actually needed)

Reply:
Confirmed that we will be able to get vm_size from IMDS. Removed this as suggested. :)